private/pkg/observability/observabilityotel/observabilityotel.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observabilityotel
+
+import (
+	"github.com/bufbuild/buf/private/pkg/observability"
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+func NewTracerProviderCloser(tracerProvider *trace.TracerProvider) observability.TracerProviderCloser {
```

Comment:
It might be nice to keep this limited to using the trace API and not sdk - i.e. take https://pkg.go.dev/go.opentelemetry.io/otel/trace#TracerProvider and then do type assertions to an interface with `Shutdown(context.Context)` to determine if we should call that on shutdown.

This keeps the buf CLI free of SDK implementation details.

Reply:
I'm a fan of this, going to have an interface here that includes `Shutdown(context.Context) error` and then we'll use that :)